### PR TITLE
Cleanup TensorType

### PR DESCRIPTION
**Pull request by gramalingam**
Merge the two classes TensorType and ParametricTensor into one class, and related cleanup.

### CODE REVIEW

File: onnxscript/onnx_types.py [hunk 1]
```diff
@@ -3,85 +3,107 @@
...
+DimType = Union[int, str, type(None)]
 
-    def __str__(self) -> str:
-        shapestr = str(self.shape) if self.shape else "[...]"
-        return onnx.TensorProto.DataType.Name(self.dtype) + shapestr
 
-    def __repr__(self) -> str:
-        return f"{self.__class__.__name__}(dtype={self.dtype!r}, shape={self.shape!r})"
+def check_dim(dim):
+    if not isinstance(dim, (int, str, type(None))):
```
Comment (justinchuby) [inline onnxscript/onnx_types.py:23]: Use DimType?

File: onnxscript/onnx_types.py [hunk 2]
```diff
@@ -3,85 +3,107 @@
...
+        raise TypeError(f"Invalid dimension {dim}")
 
-    def to_type_proto(self):
-        # TODO: handle None
-        return onnx.helper.make_tensor_type_proto(self.dtype, self.shape)
 
+ShapeType = Union[Tuple[DimType], DimType, type(Ellipsis)]
```
Comment (justinchuby) [inline onnxscript/onnx_types.py:27]:
```suggestion
ShapeType = Union[Tuple[DimType, ...], DimType, type(Ellipsis)]
```

File: onnxscript/onnx_types.py [hunk 3]
```diff
@@ -3,85 +3,107 @@
...
+
+    def __getitem__(self, shape: Optional[ShapeType]):
+        if self.shape is not None:
+            raise ValueError("Invalid usage: shape already specified.")
+        if shape is None:
+            # Treat FLOAT[NONE] as 1-dimensional tensor with unknown dimension
+            shape = (None,)
+        return TensorType(self.dtype, shape)
 
     def to_type_proto(self):
```
Comment (justinchuby) [inline onnxscript/onnx_types.py:55]: I would add a return type

File: onnxscript/onnx_types.py [hunk 4]
```diff
@@ -3,85 +3,107 @@
...
+FLOAT16 = TensorType(onnx.TensorProto.FLOAT16)
+DOUBLE = TensorType(onnx.TensorProto.DOUBLE)
+UINT32 = TensorType(onnx.TensorProto.UINT32)
+UINT64 = TensorType(onnx.TensorProto.UINT64)
+COMPLEX64 = TensorType(onnx.TensorProto.COMPLEX64)
+COMPLEX128 = TensorType(onnx.TensorProto.COMPLEX128)
+BFLOAT16 = TensorType(onnx.TensorProto.BFLOAT16)
+
+
+def onnx_type_to_os_type_repr(onnx_type: onnx.TypeProto):
```
Comment (justinchuby) [inline onnxscript/onnx_types.py:85]:
```suggestion
def onnx_type_to_os_type_repr(onnx_type: onnx.TypeProto) -> str:
```

File: onnxscript/onnx_types.py [hunk 5]
```diff
@@ -3,85 +3,107 @@
...
+    """Converts an onnx type into the string representation of the type in *onnx-script*.
+    Args:
+        onnx_type: an instance of onnx TypeProto
+
+    Returns:
+        The string representation of the type in onnx-script
```
Comment (justinchuby) [inline onnxscript/onnx_types.py:91]:
```suggestion
        The string representation of the type in onnx-script
        
    Raises:
        ...
```

File: onnxscript/onnx_types.py (hunk 4 above)
Comment (justinchuby) [inline onnxscript/onnx_types.py:85]: naming: maybe: onnx_type_to_onnxscript_repr?

File: onnxscript/onnx_types.py (hunk 2 above)
Comment (gramalingam) [reply onnxscript/onnx_types.py:27]: Good catch!

Comment (gramalingam) [reply onnxscript/onnx_types.py]: Done.

File: onnxscript/onnx_types.py (hunk 1 above)
Comment (gramalingam) [reply onnxscript/onnx_types.py:23]: Not sure how. DimType is a type-alias, used for shortening type-annotations/signature. isinstance does not support use of type-alias with generics in it.

Comment (justinchuby) [reply onnxscript/onnx_types.py:23]: SG!

**Review by justinchuby — APPROVED** [review]
LGTM🚀

**Pull request merged**